docs/rfcs/1012-parsec-auth.md

```diff
@@ -0,0 +1,432 @@
+  - The parameters used to generate the symmetric key (`AUTH_MEDIUM_SYM_KEY`) that encrypted `enc(AUTH_MEDIUM_PRIV_KEY)` (`AUTH_MEDIUM_SYM_KEY_PARAMS`)
+
+> [!WARNING]
+> A user should be able to register multiple authentication methods,
+
+### The authentication process
+
+We use a HMAC to authenticate each request to the server with the secret `AUTH_MEDIUM_HMAC_KEY` that was previously shared.
+
+### Obtain the account manifest from the server
+
+The server store the encrypted account manifest that contain the important information about the user.
+For the user to obtain the account manifest from the server it will do:
+
+1. Be authenticated with the server.
+2. Retrieve the following encrypted data from the sever:
+
+   1. Account manifest (`enc(ACCOUNT_MANIFEST)`)
+   2. Account manifest symmetric key (`enc(ACCOUNT_MANIFEST_SYM_KEY)`)
+   3. Authentication medium private key (`enc(AUTH_MEDIUM_PRIV_KEY)`)
+   4. The parameters used to generate the symmetric key that encrypt the private key (`AUTH_MEDIUM_SYM_KEY_PARAMS`)
+
+3. Re-generate `AUTH_MEDIUM_SYM_KEY` from the parameters and the password
+4. Decrypt `enc(AUTH_MEDIUM_PRIV_KEY)` with `AUTH_MEDIUM_SYM_KEY` to obtain the private key (`AUTH_MEDIUM_PRIV_KEY`)
+5. Decrypt `enc(ACCOUNT_MANIFEST_SYM_KEY)` with `AUTH_MEDIUM_PRIV_KEY` (`ACCOUNT_MANIFEST_SYM_KEY`)
+6. Decrypt `enc(ACCOUNT_MANIFEST)` with `ACCOUNT_MANIFEST_SYM_KEY`
+7. Decrypt the account manifest
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Alice
+  participant S as  Authentication server
+
+  Note over Alice,S: Alice is already authenticated to the server
+
+  S ->> Alice: Retrieve `enc(ACCOUNT_MANIFEST)`, `enc(ACCOUNT_MANIFEST_SYM_KEY)`,<br>`enc(AUTH_MEDIUM_PRIV_KEY)` & `AUTH_MEDIUM_SYM_KEY_PARAMS`
+  Alice ->> Alice: Generate `AUTH_MEDIUM_SYM_KEY` using the password and `AUTH_MEDIUM_SYM_KEY_PARAMS`
+  Alice ->> Alice: Decrypt `enc(AUTH_MEDIUM_PRIV_KEY)` with `AUTH_MEDIUM_SYM_KEY`
+  Alice ->> Alice: Decrypt `enc(ACCOUNT_MANIFEST_SYM_KEY)` with `AUTH_MEDIUM_PRIV_KEY`
+  Alice ->> Alice: Decrypt `enc(ACCOUNT_MANIFEST)` with `ACCOUNT_MANIFEST_SYM_KEY`
+```
+
+### Uploading a new device
+
+To upload a new device, the user would need to:
+
+1. [Obtain the account manifest from the server](#obtain-the-account-manifest-from-the-server).
+2. Create a new symmetric key for encrypting the device (`DEVICE_SYM_KEY`).
+3. Encrypt the new device with `DEVICE_SYM_KEY` (`enc(DEVICE)`).
+4. Add `DEVICE_SYM_KEY` to the account manifest.
+5. Encrypt the account manifest with the symmetric key `ACCOUNT_MANIFEST_SYM_KEY`.
+5. Generate a device ownership proof (`DEVICE_OWNERSHIP_PROOF`):
+
+   The proof is generated by using the signature key of the device to sign the blob `device_id` + `auth_medium_id` currently used to authenticate the request.
+
+6. Upload the organization, the encrypted device, the device ownership proof, and the updated encrypted account manifest.
```

Comment:
what do you mean by "Upload the organization" ?

Reply:
I've added precision to the organization id